js/bitopro.js

```diff
@@ -0,0 +1,1349 @@
+            'id': id,
+            'pair': market['id'],
+        };
+        const response = await this.privateDeleteOrdersPairId (this.extend (request, params));
+        //
+        //     {
+        //         "orderId":"8777138788",
+        //         "action":"SELL",
+        //         "timestamp":1644899002465,
+        //         "price":"16000",
+        //         "amount":"0.01"
+        //     }
+        //
+        return this.parseOrder (response);
+    }
+
+    async cancelAllOrders (symbol = undefined, params = {}) {
+        await this.loadMarkets ();
+        const request = {
+            // 'pair': market['id'], // optional
+        };
+        // privateDeleteOrdersAll or privateDeleteOrdersPair
+        let method = this.safeString (this.options, 'privateDeleteOrdersPair', 'privateDeleteOrdersAll');
+        if (symbol !== undefined) {
+            const market = this.market (symbol);
+            request['pair'] = market['id'];
+            method = 'privateDeleteOrdersPair';
+        }
+        const response = await this[method] (this.extend (request, params));
+        const result = this.safeValue (response, 'data', {});
+        //
+        //     {
+        //         "data":{
+        //             "BNB_TWD":[
+        //                 "9515988421",
+        //                 "4639130027"
+        //             ]
+        //         }
+        //     }
+        //
+        return result;
+    }
+
+    async fetchOrder (id, symbol = undefined, params = {}) {
+        await this.loadMarkets ();
+        if (symbol === undefined) {
+            throw new ArgumentsRequired ('fetchOrder requires the symbol parameter');
+        }
+        const market = this.market (symbol);
+        const request = {
+            'orderId': id,
+            'pair': market['id'],
+        };
+        const response = await this.privateGetOrdersPairOrderId (this.extend (request, params));
+        //
+        //     {
+        //         "id":"8777138788",
+        //         "pair":"bnb_twd",
+        //         "price":"16000",
+        //         "avgExecutionPrice":"0",
+        //         "action":"SELL",
+        //         "type":"LIMIT",
+        //         "timestamp":1644899002598,
+        //         "status":4,
+        //         "originalAmount":"0.01",
+        //         "remainingAmount":"0.01",
+        //         "executedAmount":"0",
+        //         "fee":"0",
+        //         "feeSymbol":"twd",
+        //         "bitoFee":"0",
+        //         "total":"0",
+        //         "seq":"BNBTWD548774666",
+        //         "timeInForce":"GTC",
+        //         "createdTimestamp":1644898944074,
+        //         "updatedTimestamp":1644899002598
+        //     }
+        //
+        return this.parseOrder (response);
+    }
+
+    async fetchOrders (symbol = undefined, since = undefined, limit = undefined, params = {}) {
+        await this.loadMarkets ();
+        if (symbol === undefined) {
```

Comment:
Let's move all these checks above `loadMarkets()` throughout this file and let's also style the error message uniformly with the other exchanges.